src/include/function/gds/rec_joins.h

```diff
@@ -57,19 +58,24 @@ struct RJBindData final : public GDSBindData {
     }
 };
 
+using gds_terminate_func_t =
+    std::function<bool(GDSFrontier& frontier, const processor::NodeOffsetMaskMap& maskMap)>;
+
 // Wrapper around the data that needs to be stored during the computation of a recursive joins
 // computation from one source. Also contains several initialization functions.
 struct RJCompState {
     std::unique_ptr<function::FrontierPair> frontierPair;
     std::unique_ptr<function::EdgeCompute> edgeCompute;
     std::unique_ptr<RJOutputs> outputs;
     std::unique_ptr<RJOutputWriter> outputWriter;
+    gds_terminate_func_t terminateFunc;
```

Comment:
Either is fine but I don't think creating a class is necessary because there is no state. Even if we use a class, it will not be maintained with unique_ptr because there is no inheritance.